test/unittests/test_cpp.cpp

```diff
@@ -120,3 +124,36 @@ TEST(cpp, host_call)
 
     example_host_destroy_context(host_context);
 }
+
+TEST(cpp, result_raii)
+{
+    static auto release_called = false;
```

Comment:
Do you need it to be static? Why not local and capture it in lambda?

Reply:
Because the `.release` field is coming from a C struct that only uses pointers to functions. For convenience, we use the fact you can convert non-capturing lambdas to function pointers.